### ISSUE TITLE
Introduce FAQPage as subtype (sibling?) of QAPage, for "Frequently asked questions"

### ISSUE THREAD

**Issue by danbri** [schemaorg/schemaorg#1723]
Context: when we did the Q/A design a few years ago (/cc @scor) we included a "QAPage" construct, to capture the idea that there are pages which carry collections of questions and answers. In retrospect I wasn't sure that it was a great idea, but the type is there.

Talking with colleagues at Google, we are interested to have schema that distinguishes "Frequently asked questions" content on the Web. Often these are several Q/A pairs to a page, sometimes organized into categories. Other times they are published one page per question.

We think "about" or "keywords" properties could handle the categorization aspect, and that perhaps the [itemref](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/itemref) construct might be useful to avoid duplication/verbosity when using Microdata and when all the items in one category have the same keywords. As an aside, perhaps in future "categories", "keywords" could be considered as subproperty of "about" and their structure/definition converged. 

Proposal:

Add a new type FAQPage as a subtype  [edit: or _sibling_] of QAPage, definition:

> A [[FAQPage]] is a [[QAPage]] presenting one or more "[Frequently asked questions](https://en.wikipedia.org/wiki/FAQ)".

We would add examples for a single page with a single question, and for a larger page with several questions, including categories handled via "about" and/or "keywords" properties on the QAPage and/or the Question (assuming that Answers rarely get dedicated metadata for such things).

Pinging @scor @tmarshbing @nicolastorzec @vholland @rvguha @tilid for perspectives

**Comment by rvguha** [comment]
Looks good to me.

guha

**Comment by vholland** [comment]
Looks good to me

**Comment by nicolastorzec** [comment]
I like the idea of having FAQs but I don't see the benefit of making it a subclass of [QAPage](http://schema.org/QAPage). 

- A QA page from a web portal like Quora, Yahoo Answer, or StackExchange is typically about **one** question and its answer(s) from one (or more) user(s).

- A FAQ page is typically a collection of curated <question, answer> pairs, potentially organized into topical sections: i.e. multiple sections/topics per FAQ, multiple questions per section/topic, one official answer per question, one curator, etc.

FAQPage and QAPage could just be siblings?

**Comment by emekaokoye** [comment]
FAQPage is a very good idea. 

But let FAQPage be a subclass of  schema:webpage while QAPage be a subclass of FAQPage. 

Why? 

Simply because QAPage represents a schema:webpage with multiple answers and FAQPage represents one of the following:
1. Schema:webpage with 1 question and N answers. 
2.  Schema:webpage with N questions and N answers. 
3.  Schema:webpage with 1 question and 1 answer. 
4.  Schema:webpage with N questions and 0 answers but questions are linked to #3.

**Comment by MrJonPayne** [comment]
I'd be careful about the 'keywords' & 'category' properties. You will likely find a lot of excitable SEO chaps stuffing these with poor quality data in the mistaken belief that they will somehow achieve some sort of answer box nirvana on Google search results pages.

**Comment by VladimirAlexiev** [comment]
- isn't there a general mechanism to indicate preferredness, notability, or importance? If not, I think we need one.
I'm thinking of cultural heritage, where it's important to indicate the preferred image and title of an artwork
- @emekaokoye @nicolastorzec  QAPage is defined as "focussed on a **specific** Question and its Answer(s)" while FAQPage is only about important questions, so indeed these would be siblings.
- please change the title to say "subtype of QAPage" (now it says subtype of itself)

**Comment by danbri** [comment]
@MrJonPayne indeed, a lot of people see 'keywords' as a kind of 1990s relic. But it may yet still have usecases when carefully used.

**Comment by danbri** [comment]
Ok, the point about sibling rather than subtype is sounding persuasive; I've amended the title here for now

**Comment by Aaranged** [comment]
I like where this is going.  In particular, I think planning to accommodate FAQs that consist of multiple question/answer pairs is critical, as this is probably the most common form of FAQ pages.

I would think it's obvious, but one of the things that this type should leverage is existing Question/question and Answer/answer vocabulary.  ItemList may also have a place in being able to address those multiple Q/A pairs

Broadly agree that this makes more sense as a sibling than a subtype.

**Comment by nicolastorzec** [comment]
@VladimirAlexiev: I added some thoughts on preferredness/notability/importance on https://github.com/schemaorg/schemaorg/issues/1726

**Comment by kidehen** [comment]
@danbri ,

Here's our FAQ Ontology : http://www.openlinksw.com/ontology/faq# .

**Comment by jeannieh** [comment]
With Google Maps now offering a means to both ask and answer questions, this may possibly be confusing if it is made a subclass of QAPage. I too, agree with the thought about using a sibling.

**Comment by danbri** [comment]
Ok, draft staged at http://pending.webschemas.org/FAQPage - please take a look

(Once we're happy let's cross-reference it back from QAPage too.)

**Comment by danbri** [comment]
Note that http://pending.schema.org/BackgroundNewsArticle also talks about FAQs. We should figure out how these relate.

**Comment by danbri** [comment]
I wouldn't say FAQPage is a subtype of BackgroundNewsArticle as not all FAQs are news-related. But we should at least cross-link the definitions.

**Comment by danbri** [comment]
Any final comments on this? It is on its way to be included in the pending section of our next release

**Comment by magicomartinez** [comment]
Looks good. Would it be useful to add a reference to schema.org/Question? Something along the lines of...

> A [FAQPage](http://pending.webschemas.org/FAQPage) is a [WebPage](http://pending.webschemas.org/WebPage) presenting one or more frequently asked [Question](http://schema.org/Question) (see also [QAPage](http://pending.webschemas.org/QAPage)).

**Comment by AnhellO** [comment]
> Is it planned to be ever released? Seems around a year have passed.
> Is it possible to use QAPage for google to be indexed by? With more then one question+answer on page, or would it just reject all questions but one?

@Mart-Bogdan, I am still waiting for the FAQ schema news as well. I am using it in the meantime, even when it is still being under the pending schema, because the Q&A schema markup won't work for pages with more than one question, you will get multiple warnings on the structured data tool, even if you try it:

* https://searchengineland.com/google-launches-qa-structured-data-for-question-answer-pages-308999
* https://developers.google.com/search/docs/data-types/qapage
  * > Don't use QAPage markup for FAQ pages or pages where there are multiple questions per page. QAPage markup is for pages where the focus of the page is a single question and its answers.

**Comment by Mart-Bogdan** [comment]
@AnhellO actually there are no errors/warning in test tool, but I believe google would silently ignore it.

![image](https://user-images.githubusercontent.com/775038/49815851-58f12600-fd75-11e8-90cd-6875001b8413.png)

**Comment by AnhellO** [comment]
That makes sense, maybe they applied those changes to the structured data tool with the recent updates they did to the QAPage schema markup. In any case, I guess for both cases it will silently ignore (FAQPage and QAPage with multiple questions).

**Comment by chaduhduh** [comment]
I would benefit from a FAQPage schema as well separate from the QAPage schema. Any updates on this??

**Comment by ayumihamsaki** [comment]
@danbri I was looking at the pending spec would it not be a good idea to add some special properties to that class for example:

- question
- answer

Also create an extra `@type` so people can mark up a list of Question and Answers.

Something like:

```html
<script type="application/ld+json">
{
 "@context": "http://schema.org",
 "@type": "FAQPage",
 "itemListElement":
 [
  {
   "@type": "ListItem",
   "position": 1,
   "item":
   {
    "question": "what color widgets to you sell?",
    "answer": "we sell blue and red ones"
    }
  },
  {
   "@type": "ListItem",
  "position": 2,
  "item":
   {
     "qusetion": "how much do the widgets cost",
     "answer": "1usd each"
   }
  }
 ]
}
</script>
```

**Comment by ayumihamsaki** [comment]
@danbri I was looking at the pending spec would it not be a good idea to add some special properties to that class for example:

- question
- answer

Also create an extra `@type` so people can mark up a list of Question and Answers.

Something like:

```html
<script type="application/ld+json">
{
 "@context": "http://schema.org",
 "@type": "FAQPage",
 "name": "FAQ on Widgets",
 "description": "Find answers to the most popular questions about our range of widgets",
 "url": "https://www.example.com/faq",
 "numberOfItems": 2,
 "itemListElement":
 [
  {
   "@type": "ListItem",
   "position": 1,
   "item":
   {
    "question": "what color widgets do you sell?",
    "answer": "we sell blue and red ones"
    }
  },
  {
   "@type": "ListItem",
  "position": 2,
  "item":
   {
     "qusetion": "how much do the widgets cost?",
     "answer": "1usd each"
   }
  }
 ]
}
</script>
```

**Comment by dbiollo** [comment]
A sibling type QAPage is documented in Google Structured Docs as having "mainEntity" where the Question & Answers go: https://developers.google.com/search/docs/data-types/qapage#QAPage

schema.org/Question and schema.org/Answer already encapsulate questions and answers.

The same could work for FAQPage, as mainEntity can be a repeated field, containing Question with an Answer.

**Comment by ayumihamsaki** [comment]
@dbiollo Thanks for the info, I didn't realize that.

Maybe the website should be updated, the following web pages:

https://pending.schema.org/FAQPage
https://schema.org/QAPage

Both have the section "**Properties from CreativeWork**" but the properties:

https://schema.org/Answer
https://schema.org/Question

Are not listed - it would be helpful to have them listed under the "**Properties from CreativeWork** on the web pages `FAQPage` and `QAPage` - instead of hidden.

**Comment by MichaelAndrews-RM** [comment]
@ayumihamsaki The schema.org vocabulary already has entities for [Question](https://schema.org/Question) (“A specific question - e.g. from a user seeking answers online, or collected in a Frequently Asked Questions (FAQ) document.”) and [Answer](https://schema.org/Answer).   

The relationship between Question, Answer, and FAQPage should be explicit in the documentation so that webmasters know how to use them together.

**Comment by ayumihamsaki** [comment]
Thanks both for your answers. I would like to re-rite my comment to this:

The problem is that Schema.org is the '**_Spec_**' and Google uses the Spec (not Schema using the Google Developers infomation). Everyone is linking to the Google Developers Page here: https://developers.google.com/search/docs/data-types/qapage

Yet the Schema.org webpages found here:

https://schema.org/QAPage
https://pending.schema.org/FAQPage

There is no mention under the section "**_Properties from CreativeWork_**" listing the properties:

https://schema.org/Question
https://schema.org/Answer
etc.

I suggest the schema team add the extra information.

Also I have one last question, `schema:QAPage` Spec says to use only ONE QUESTION and `schema:FAQPage` you can use multiple questions!

I would like to see an example code with multiple questions please.

**Comment by MichaelAndrews-RM** [comment]
@ayumihamsaki Good questions.  I agree that the community documentation needs more detailed examples -- I think we are drawing on the Google example of QAPage because an example in schema.org's documentation for an FAQPage is lacking an example.  In the schema.org spec, there are no recommended properties, so the publisher needs to consider how consumers will use the data.  I won't expect suggestedAnswers and upvoteCounts would be used.  

I would expect a FAQPage to have multiple questions.  @dbiollo says a mainEntity can be repeated (though the definition of mainEntity in the documentation reads in the single: "the primary entity described in some page")  If mainEntity can handle more than one question, I would create an array and list all the Question & Answer statements within the array.  Someone more expert than me can correct me about that approach!

**Comment by ayumihamsaki** [comment]
@MichaelAndrews-RM Thanks for the extra info! Didn't know that the `mainEntity` can be repeated.

**Comment by riazhamed143** [comment]
Any comments on the below code as it is working fine with Structure Data Testing Tool. No Errors or warning

<script type="application/ld+json">
{
 "@context": "http://schema.org/",
 "@type": "FAQPage",
 "@graph": [
 {
       "@type": "Question",
       "text": "What is a cat?",
       "dateCreated": "2019-01-27T14:01Z",
       "acceptedAnswer": {
           "@type": "Answer",
           "text": "(The text of the accepted answer goes here...).",
           "dateCreated": "2019-01-27T14:01Z"
       }
   },
   
    {
       "@type": "Question",
       "text": "Who is the best play in the world?",
       "dateCreated": "2019-01-27T14:01Z",
       "acceptedAnswer": {
           "@type": "Answer",
           "text": "Cristiano Ronaldo",
           "dateCreated": "2019-01-27T14:01Z"
       }
   }
	]
}
</script>



![image](https://user-images.githubusercontent.com/33475690/51801090-a1e03980-2252-11e9-864b-136e0f6e2a6e.png)

**Comment by Ezra-Siton-UIX** [comment]
Please add this feature (Usefull for endless sites). The QA is to complex (Most sites without this data structure). 

> Something like:
> 
> ```
> <script type="application/ld+json">
> {
>  "@context": "http://schema.org",
>  "@type": "FAQPage",
>  "name": "FAQ on Widgets",
>  "description": "Find answers to the most popular questions about our range of widgets",
>  "url": "https://www.example.com/faq",
>  "numberOfItems": 2,
>  "itemListElement
> ```

**Comment by chaduhduh** [comment]
@riazhamed143 , I believe @Mart-Bogdan (and others) suggested that although there are no errors or warnings (with the provided snippet), it will simply be ignored by Google. Not sure if that is the info you are after or not.

**Comment by danbri** [comment]
This got deployed (e.g. see [this article](https://searchengineland.com/google-confirms-new-faqs-qa-and-how-tos-for-search-results-snippets-303116)). Let's move it into core.

**Comment by DamonHD** [comment]
How is one meant to deal with pages which embed a FAQ/Q&A section, eg within an article or review?

**Comment by DamonHD** [comment]
To answer my own question, though I have no idea whether this is a Good Idea, is that I have made a typical such page BOTH an Article and a FAQPage (ie itemtype has both), so that it can have features of both.  (Yandex and Google checkers seem happy with this, for example.)

**Comment by RichardWallis** [comment]
Implemented in [release 3.5 ](https://schema.org/docs/releases.html#v3.5)
